### PR TITLE
[Bloganuary] Modal analytics tracking

### PR DESCRIPTION
**Pull request by thomashorta**
Part of #19685

Add tracking for the Bloganuary Learn More model based on specs (pctCYC-16J-p2).

-----

## To Test:

1. Install and log into the app with a site that has Blogging Prompts available
2. Go to the My Site dashboard
3. Tap "Learn More" on the Bloganuary card
4. **Verify**, in the logs, `🔵 Tracked: bloganuary_nudge_learn_more_modal_shown, Properties: {"prompts_enabled":[true if prompts are enabled, false otherwise]}`
8. Tap the CTA action
9. **Verify**, in the logs, `🔵 Tracked: bloganuary_nudge_learn_more_modal_action_tapped, Properties: {"action":["dismiss" or "turn_on_prompts", depending if prompts are enabled or not]}`

Modal dismissal:
Whenever the Modal is dismissed (no matter which action led to it), there should be tracking for dismissal. Test multiple dismiss methods (back button, close button, fling down, tap CTA) and **verify** `🔵 Tracked: bloganuary_nudge_learn_more_modal_dismissed` in the app logs.

-----

## Regression Notes

1. Potential unintended areas of impact

    - N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - N/A

3. What automated tests I added (or what prevented me from doing so)

    - Analytics tracker unit tests.

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## UI Changes Testing Checklist:
No UI changes.

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/test/java/org/wordpress/android/ui/bloganuary/BloganuaryNudgeAnalyticsTrackerTest.kt [hunk 1]
```diff
@@ -58,4 +59,37 @@ class BloganuaryNudgeAnalyticsTrackerTest : BaseUnitTest() {
             )
         }
     }
+
+    @Test
+    fun `WHEN trackLearnMoreOverlayShown is called THEN analyticsTracker is called correctly`() {
+        listOf(true, false).forEach { isPromptsEnabled ->
```
Comment (RenanLukas) [inline WordPress/src/test/java/org/wordpress/android/ui/bloganuary/BloganuaryNudgeAnalyticsTrackerTest.kt:63]: 👍

File: WordPress/src/test/java/org/wordpress/android/ui/bloganuary/BloganuaryNudgeAnalyticsTrackerTest.kt [hunk 2]
```diff
@@ -58,4 +59,37 @@ class BloganuaryNudgeAnalyticsTrackerTest : BaseUnitTest() {
...
+        tracker.trackLearnMoreOverlayDismissed()
+
+        verify(analyticsTracker).track(
+            Stat.BLOGANUARY_NUDGE_LEARN_MORE_MODAL_DISMISSED
+        )
+    }
+
+    @Test
+    fun `WHEN trackLearnMoreOverlayActionTapped is called THEN analyticsTracker is called correctly`() {
+        BloganuaryNudgeLearnMoreOverlayAction.entries.forEach {
```
Comment (RenanLukas) [inline WordPress/src/test/java/org/wordpress/android/ui/bloganuary/BloganuaryNudgeAnalyticsTrackerTest.kt]: 👍

**Review by RenanLukas — APPROVED** [review]
Thanks for the PR. I like what you did with the unit tests 👍 

It's working as expected and code LGTM.

File: WordPress/src/test/java/org/wordpress/android/ui/bloganuary/BloganuaryNudgeAnalyticsTrackerTest.kt (hunk 2 above)
Comment (thomashorta) [reply WordPress/src/test/java/org/wordpress/android/ui/bloganuary/BloganuaryNudgeAnalyticsTrackerTest.kt]: Based on your comment here (https://github.com/wordpress-mobile/WordPress-Android/pull/19704#discussion_r1411410090) I will actually change this test to test all possibilities manually and compare against hard-coded expected values so we get failures if the enum is changed by mistake.

**Pull request merged**